pTables.py

```diff
@@ -0,0 +1,235 @@
+            self._t.del_row(row_number)
+
+
+    def output_table(self):
+        """
+        prints the object form of the table
+        """
+        if not self.is_table_empty():
+            print(self._t)
+
+
+    def get_string_version(self):
+        """
+        :return: The string variable of the table representation.
+        """
+        if not self.is_table_empty():
+            return self._t.get_string()
+
+
+    def sort_table(self, s):
+        """
+        function checks whether or not the string passed is in the header of the table or not. If it is not,
+        the function does nothing.
+        :param s: String passed by the caller. If not a string when classed, function converts (s) to a string.
+        :return: the string representation of the table object, sorted by the string passed by caller.
+        """
+        if not self.is_table_empty():
+            s = str(s)
+            if s in self.header:
+                return self._t.get_string(sortBy=s)
+
+
+
+########################################################################################################################
+########################################################################################################################
+############################# Private functions to class: MyTable    Please No Touch ###################################
+########################################################################################################################
+########################################################################################################################
+
+    def _get_data_row_length(self):
+        """
+        Private function to class MyTable.
+
+        Function assigns the member variable self.dataRowLength with the length of all rows in the table
+        containing the data. This length excludes the header of the table, self.matrixTable[0]
+        """
+
+        if not self.is_table_empty():
+            self.dataRowLength = len(self.matrixTable[1:])
+
+
+    def _put_data(self):
+
+        """
+         Private function to class MyTable.
+         Function installs the data in the table passed to this table objects constructor.
+        :return:
+        """
+        if not self.is_table_empty():
+            self._t.field_names = self.header = self.matrixTable[0]
+            [self._t.add_row(i) for i in self.matrixTable[1:]]
+            self._get_max_column_lengths()
+
+    def _get_max_column_lengths(self):
+        """
+         Private function to class MyTable.
+         Function loops through an calculates the maximum length of each column in the table. It does this by
+         finding the longest string present in any column in the table. Then installs this value in the member variable
+         self.columnWidth array. Where each number in the array corresponds to that columns max width.
+        :return:
+        """
+        if not self.is_table_empty():
```

Comment:
If table is empty, what should it put in this variable. Also, if this fucntion is ran twice would it add all of the data to the list twice?

Reply:
Yes it will do it twice if call twice. Being a private function though, this function will only be called once, when the table is initially created.